onnxruntime/core/framework/execution_context.cc

```diff
@@ -15,6 +15,7 @@ class DeviceStreamCollectionImpl {
     for (auto& ep : providers) {
       eps_.push_back(ep);
     }
+    under_subgraph_ = sess_state.GetGraphViewer().ParentNode() != nullptr;
```

Comment:
>under_subgraph_ [](http://example.com/codeflow?start=4&length=15)

The logic is impeccable while could be more intuitive - instead of under_subgraph_ there could be a default false flag - "is_main_graph", and initialize by:

is_main_graph = ... == nullptr

in destructor, we do:

if (is_main_graph) { flush...}

The principal here is to avoid using an extra ! if we can.